### PR TITLE
#1987 Adding CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Bakaniko**
WIP to clode this issue:
https://trac.osgeo.org/osgeolive/ticket/1987

Following mail thread:
https://lists.osgeo.org/pipermail/osgeolive/2018-September/013566.html

### CODE REVIEW

**Comment by cvvergara** [comment]
Very challenging task you started :)

**Comment by cvvergara** [comment]
I added as many reviwers as I could think of due to the sensitivity of the PR.
As long as there is  TODO, I can not agree on the merge.
If I want to make a change I will do it by pushing to the adding_contributing.md branch on Bakaniko/OSGeoLive-doc.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via [issue on our bug tracker](https://trac.osgeo.org/osgeolive/report/1),
+[OSGeoLive mailing list](https://lists.osgeo.org/mailman/listinfo/osgeolive), or any other method with the owners of this repository before making a change. 
+
+By submitting you are agreeing to make your content available via our [open license policy](https://live.osgeo.org/en/copyright.html).
+
+In the same manner, when translating on Transifex, you are agreeing to make your translations following those licence terms.
+
+Please note we have a code of conduct, please follow it in all your interactions with the project.
```
Comment (cvvergara) [inline CONTRIBUTING.md]: Missing link to code of conduct.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+1. TO-DO
+
+## Translation process
+### Join a translation team and translate
+
+1. Create an account or login on [Transifex](https://www.transifex.com/osgeo/osgeolive/)
+2. Request joining a translation team
+3. Translate ! 
+4. Please consider adding your self to our [contributor list](https://github.com/OSGeo/OSGeoLive-doc/blob/master/contributors.csv), if you don't know how to use git, please contact a maintainer.
+
```
Comment (cvvergara) [inline CONTRIBUTING.md]: Right now we have the egg-hen problem.
egg = sign contirbution agreement
hen = start translating, we will use your translations

the hen is already given, but we didnt have the egg

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+### Request a new language
+
+1. TO-DO
+
+## Code of Conduct
+
+**Bakaniko's comment:** This part was part of the template, it can be
```
Comment (cvvergara) [inline CONTRIBUTING.md]: Can be in a separate file

File: CONTRIBUTING.md (hunk 1 above)
Comment (cvvergara) [reply CONTRIBUTING.md]: Hmm, too many things in one file, (just saw the code of conduct down)

**Comment by cvvergara** [comment]
I wonder, because we have 2 repositories, what ever is on one repo, must be also on the other repo

**Comment by cvvergara** [comment]
Contributions via mail for example, I remember last year Steve made a contribution via mail that bakaniko added to the repo. What if Steve has had not signed the contirbution agreement, then bakaniko could not being able to use that contribution.
Changes to the wiki's are considered contributions? if so, what license do they have?

**Comment by bencaradocdavies** [comment]
It would be interesting to compare the OSGeo code of conduct with the [Contributor Covenant](https://www.contributor-covenant.org/). The Linux kernel community has recently adopted a version of the covenant.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -12,7 +12,76 @@ Please note we have a code of conduct, please follow it in all your interactions
...
+* Extensive or creative comments.
+
+### CLA
+A contributor licensing agreement (CLA) must be filled out by every contributor to OSGeoLive.
+
+The CLA makes everyone’s rights clear:
+
+You (or your company) grant copyright license for your contributions to CARTO.
+You (or your company) grant patent license for your contributions to CARTO.
+The contribution is entirely voluntary.
```
[the remark this answers is not in the document]
Comment (Bakaniko) [reply CONTRIBUTING.md]: I'm not sure to understand your point of view correctly.
I'll wait @cvvergara , @kalxas, @camerons point of view on that.

Maybe @astroidex and @darkblue-b as well

**Comment by astroidex** [comment]
@bakaniko @cvvergara shall we take this file to 13.0 version? Still afterwards changes can be done if needed.

**Comment by Bakaniko** [comment]
@astroidex , maybe the simple way is the one suggested by @bencaradocdavies : refer to OSGeo resources.
Let's talk about it next meeting

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
+# Contributing
+
+When contributing to this repository, read the Pull Request Process section and if necessary,discuss the change you wish to make via [issue on our bug tracker](https://trac.osgeo.org/osgeolive/report/1),
```
Comment (neteler) [inline CONTRIBUTING.md]:
```suggestion
When contributing to this repository, read the Pull Request Process section and if necessary, discuss the change you wish to make via [issue on our bug tracker](https://trac.osgeo.org/osgeolive/report/1),
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+## Pull Request Process
+
+1. Evaluate the change
+2. If necessary, sign the **contributor licensing agreement**
+3. For big changes fill, eventually engage discussion in a new issue or in the mailing list.
+4. Make the PR
+
+### Evaluate the change
+
+In the OSGeoLive, we see 2 type of contribution.
```
Comment (neteler) [inline CONTRIBUTING.md]:
```suggestion
In the OSGeoLive, we see two types of contribution.
```

File: CONTRIBUTING.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (astroidex) [reply CONTRIBUTING.md:60]: CLA - do we ask contributors to sign a CLA at the moment?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  Fix typo in install file docs.
+  Obvious fix.
+
+## Translation process
+### Join a translation team and translate
+
+1. Create an account or login on [Transifex](https://www.transifex.com/osgeo/osgeolive/)
+2. Request joining a translation team
+3. Translate ! 
+4. Please consider adding your self to our [contributor list](https://github.com/OSGeo/OSGeoLive-doc/blob/master/contributors.csv), if you don't know how to use git, please contact a maintainer.
```
Comment (astroidex) [inline CONTRIBUTING.md:89]: yourself  not your self

**Review by astroidex — COMMENTED** [review]
Looks good to me. See comments.

**Comment by kalxas** [comment]
Hi all, we are still missing a contribution guide, can we please review and merge?

File: CONTRIBUTING.md (hunk 7 above)
Comment (neteler) [inline CONTRIBUTING.md:89]:
```suggestion
4. Please consider adding yourself to our [contributor list](https://github.com/OSGeo/OSGeoLive-doc/blob/master/contributors.csv), if you don't know how to use git, please contact a maintainer.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (neteler) [reply CONTRIBUTING.md:10]: > Missing link to code of conduct.

Shall it be this one? https://www.osgeo.org/code_of_conduct/

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
+# Contributing
+
+When contributing to this repository, read the Pull Request Process section and if necessary, discuss the change you wish to make via [issue on our bug tracker](https://trac.osgeo.org/osgeolive/report/1),
+[OSGeoLive mailing list](https://lists.osgeo.org/mailman/listinfo/osgeolive), or any other method with the owners of this repository before making a change. 
+
+By submitting you are agreeing to make your content available via our [open license policy](https://live.osgeo.org/en/copyright.html).
+
+In the same manner, when translating on Transifex, you are agreeing to make your translations following those licence terms.
```
Comment (neteler) [inline CONTRIBUTING.md:8]:
```suggestion
In the same manner, when translating on [Transifex](https://explore.transifex.com/osgeo/osgeolive/), you are agreeing to make your translations following those licence terms.
```

**Review by astroidex — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (kalxas) [reply CONTRIBUTING.md:10]: yes

Comment (kalxas) [reply CONTRIBUTING.md:10]: I will close this one and merge. We can make improvement on another PR

**Review by kalxas — APPROVED** [review]

**Pull request merged**